### PR TITLE
SDK Generation automation

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 1]
```diff
@@ -122,11 +122,91 @@ function New-DataPlanePackageFolder() {
...
     Set-Location $sdkfolder/src
     dotnet build /t:GenerateCode
-}
\ No newline at end of file
+}
+
+function Get-ResourceProviderFromReadme($readmeFile) {
+    $readmeFileRegex = "(?<specName>.*)/resource-manager/readme.md"
+    try
+     {
```
Comment (benbp) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:199]: Nit: Slightly over-indented

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 2]
```diff
@@ -122,11 +122,91 @@ function New-DataPlanePackageFolder() {
...
+          if ($readmeFile -match $readmeFileRegex)
+          {
+              return $matches["specName"]
+          }
+      }
+      catch
+      {
+          Write-Error "Error parsing reademe info"
```
Comment (benbp) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:207]:
```suggestion
          Write-Error "Error parsing readme info"
```

File: eng/scripts/automation/Invoke-GenerateAndBuild.ps1 [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+Write-Host "projectFolder:$projectFolder"
+Remove-Item $newpackageoutput
+
+Invoke-Generate -sdkfolder $projectFolder
+if ( $? -ne $True) {
+  Write-Error "Failed to generate sdk. exit code: $?"
+  exit 1
+}
+$outputJson = [PSCustomObject]@{
+    packages = @([pscustomobject]@{packageName="$packageName"; result='succeeded'; path=@("$path");packageFolder="$path"})
```
Comment (benbp) [inline eng/scripts/automation/Invoke-GenerateAndBuild.ps1:49]: Nit: I think the PSCustomObject casting is unnecessary here if you want to save a little space.

File: eng/scripts/automation/generate-and-build.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+if [ -z $1 ]; then
+    echo "Please input inputfile"
+    exit 1
```
Comment (benbp) [inline eng/scripts/automation/generate-and-build.sh:5]:
```suggestion
    echo "Please input inputfile"
    echo "Usage: generate-and-build.sh <inputfile> <outputfile>
    exit 1
```

File: eng/scripts/automation/generate-and-build.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+fi
+
+if [ -z $2 ]; then
+    echo "Please input outputfile"
+    exit 1
+fi
```
Comment (benbp) [inline eng/scripts/automation/generate-and-build.sh:11]:
```suggestion
    echo "Please input outputfile"
    echo "Usage: generate-and-build.sh <inputfile> <outputfile>
    exit 1
```

File: eng/scripts/automation/init.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+if [ -z $1 ]; then
+    echo "Please input outputfile"
+    exit 1
```
Comment (benbp) [inline eng/scripts/automation/init.sh:5]:
```suggestion
    echo "Please input outputfile"
    echo "Usage: init.sh <inputfile>
    exit 1
```

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 7]
```diff
@@ -122,11 +122,91 @@ function New-DataPlanePackageFolder() {
...
+          Write-Error $_
```
Comment (benbp) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:208]: Did you want to re-throw here?

File: eng/scripts/automation/GenerateAndBuildLib.ps1 [hunk 8]
```diff
@@ -122,11 +122,91 @@ function New-DataPlanePackageFolder() {
...
+      }
+      Write-Host "Cannot find resouce provider info"
```
Comment (benbp) [inline eng/scripts/automation/GenerateAndBuildLib.ps1:210]:
```suggestion
      Write-Host "Cannot find resource provider info"
```

**Comment by benbp** [comment]
A few nits but otherwise the scripts look fine. I can't comment on the intended functionality of the code though as I don't have much context there.

**Review by benbp — APPROVED** [review]

File: eng/scripts/automation/GenerateAndBuildLib.ps1 (hunk 7 above)
Comment (chunyu3) [reply eng/scripts/automation/GenerateAndBuildLib.ps1]: don't need throw here

**Pull request merged**